e2e/main.e2e.spec.ts

```diff
@@ -0,0 +1,81 @@
+import path from 'path';
+import { Application, SpectronClient } from 'spectron';
+import { BrowserWindow } from 'electron';
+import NavBarPage from './pages/navbar';
+import GeneralPage from './pages/general';
+import KubernetesPage from './pages/kubernetes';
+import PortForwardingPage from './pages/portforwarding';
+import ImagesPage from './pages/images';
+import TroubleshootingPage from './pages/troubleshooting';
+const electronPath = require('electron');
+
+jest.setTimeout(1_000_000);
+
+describe('Rancher Desktop', () => {
+  let app:Application;
+  let client: SpectronClient;
+  let browserWindow: BrowserWindow;
+  let navBarPage: NavBarPage;
+  let generalPage: GeneralPage;
+  let kubernetesPage: KubernetesPage;
+  let portForwardingPage: PortForwardingPage;
+  let imagesPage: ImagesPage;
+  let troubleShootingPage: TroubleshootingPage;
+
+  beforeAll(async() => {
+    app = new Application({
+      path:             electronPath as any,
+      args:             [path.join(__dirname, '..')],
+      webdriverOptions: {},
+      env:              { NODE_ENV: 'test' }
+    });
+
+    await app.start();
+    client = app.client;
+    browserWindow = app.browserWindow;
+    navBarPage = new NavBarPage(app);
+  });
+
+  afterAll(async() => {
+    if (app && app.isRunning()) {
+      await app.stop();
+    }
+  });
+
+  it('opens the window', async() => {
+    await client.waitUntilWindowLoaded();
+    const title = await browserWindow.getTitle();
```

Comment:
yeah, eventhough typescript complains the `await` seems to be having in effect during runtime. Without the `await` the `getTitle()` returns null and the test fails. I have tried to set some hardcoded wait and tried other approaches to get the title but none of them worked. So, I would like to leave the `await` there for now. I have added a comment in the code so that we can take a look into it when we extend the e2e setup for more real test cases.

Reply:
Ah, you're right… `getTitle` actually returns a promise, but the type information is incorrect.